bundles/org.openhab.binding.modbus.lambda/src/main/java/org/openhab/binding/modbus/lambda/internal/handler/HeatingCircuitHandler.java

```diff
@@ -0,0 +1,598 @@
+                    switch (channelUID.getIdWithoutGroup()) {
+                        case CHANNEL_HEATING_CIRCUIT_OFFSET_FLOW_LINE_TEMPERATURE:
+                            writeInt16(reg50baseadress, getScaledInt16Value(command));
+                            break;
+                        case CHANNEL_HEATING_CIRCUIT_ROOM_HEATING_TEMPERATURE:
+                            writeInt16(reg50baseadress + 1, getScaledInt16Value(command));
+                            break;
+                        case CHANNEL_HEATING_CIRCUIT_ROOM_COOLING_TEMPERATURE:
+                            writeInt16(reg50baseadress + 2, getScaledInt16Value(command));
+                            break;
+
+                    }
+
+                }
+
+            } catch (LambdaException error) {
+                if (hasConfigurationError() || getThing().getStatus() == ThingStatus.OFFLINE) {
+                    return;
+                }
+                String cls = error.getClass().getName();
+                String msg = error.getMessage();
+
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        String.format("Error with: %s: %s", cls, msg));
+            }
+        }
+    }
+
+    /**
+     * Initialization: Load the config object of the block Connect to the slave
+     * bridge Start the periodic polling
+     */
+    @Override
+    public void initialize() {
+        config = getConfigAs(HeatingCircuitConfiguration.class);
+
+        logger.debug("Initializing thing with properties: {}", thing.getProperties());
+
+        startUp();
+    }
+
+    /**
+     * Adresses for the polling registers, used for reading and writing
+     */
+    private int baseadress;
+    private int reg50baseadress;
+
+    /**
+     * This method starts the operation of this handler Connect to the slave bridge
+     * Start the periodic polling1
+     */
+    private void startUp() {
+        if (comms != null) {
+            return;
+        }
+
+        ModbusEndpointThingHandler slaveEndpointThingHandler = getEndpointThingHandler();
+        if (slaveEndpointThingHandler == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE, "Bridge is offline");
+            return;
+        }
+
+        try {
+            slaveId = slaveEndpointThingHandler.getSlaveId();
+
+            comms = slaveEndpointThingHandler.getCommunicationInterface();
+        } catch (EndpointNotInitializedException e) {
+            logger.debug("HeatingCircuit: Error setting up SlaveId");
+
+        }
+
+        // if (comms == null) {
+        // @SuppressWarnings("null")
+        // String label = Optional.ofNullable(getBridge()).map(b -> b.getLabel()).orElse("<null>");
+        // updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.BRIDGE_OFFLINE,
+        // String.format("Bridge '%s' not completely initialized", label));
+        // return;
+        // }
+
```

Comment:
RDrop or use, please check opthers to. This seems a copy/paste to all handlers.

Reply:
Deleted in all handlers.